samples/DataImporter/GraphEngine.DataImporter/Importer.cs

```diff
@@ -216,33 +218,35 @@ private static void _Import<T>(IImporter<T> importer, string file, string typena
                  }
                  catch (Exception e)
                  {
-                     //Log.WriteLine(LogLevel.Error, "An error occured during import: \n{0}\n", e.Message);
+                     Log.WriteLine(LogLevel.Error, "An error occured during import: \n{0}\n", e.Message);
                  }
              });
         }
 
         public static IImporter GetImporter(string path)
         {
-            string filename = Path.GetFileName(path);
-            string filetype = Path.GetExtension(filename).ToLower();
-
-            if (filetype == ".gz" || filetype == ".zip")
+            IEnumerable<string> text = FileReader.ReadFile(path);
+            string headerRow = text.First();
+            JObject jobj = null;
+            try
             {
-                filetype = Path.GetExtension(Path.GetFileNameWithoutExtension(filename)).ToLower();
+                jobj = JObject.Parse(headerRow);
+                return new JsonImporter();
             }
-
-            switch (filetype)
+            catch(Exception)
             {
-                case ".json":
-                case ".txt":
-                    return new JsonImporter();
-                case ".csv":
-                case ".tsv":
-                    return new CsvImporter();
-                case ".ntriples":
+                if (headerRow.Count(c => c == ',') >= headerRow.Count(c => c == '\t')&&headerRow.Count(c => c == ',') >= headerRow.Count(c => c == '/'))
```

Comment:
Try move the probe for other file types out of the exception handler.
Also, we cannot completely rely on this because csv headline may contain more tabs than commas. Also ntriples should not be a default fallback. Wrapping it up: file suffix is still important. Only guess by content when we can't tell the type by suffix alone.